### ISSUE TITLE
Add `no-padding` property to `block`

### ISSUE THREAD

**Issue by TheBlueDog** [Esri/calcite-components#4116]
### Description

There are many places where we use calcite-block inside of leading and trailing panels. It's not always desirable to have the built in padding. When using other components, such as lists, inside a block the vertical rhythm and alignment is thrown off.

Modal has a property for `no-padding` that work well in order to remove the padding. The default setting could be `true` as to not break current implementations. Setting it to `false` would remove the paddings.

### Acceptance Criteria

- [ ] calcite-block has a new property added for `no-padding`
- [ ] Setting options are `true` and `false`
- [ ] Default setting of `false` maintains current behavior
- [ ] Setting of `true` removes the padding from all four sides 

### Relevant Info

This component is in use in Field Maps today and the enhancement would be utilized as soon as it is implemented.

### Which Component

Calcite-Block

### Example Use Case

Default layout with spacing:

![Screen Shot 2022-02-15 at 1 58 37 PM](https://user-images.githubusercontent.com/5419306/154134213-a3eedf08-51af-4526-90b2-3359d66c5a67.png)

Layout with spacing removed:

![Screen Shot 2022-02-15 at 1 59 25 PM](https://user-images.githubusercontent.com/5419306/154134311-171ee308-b213-4e49-b488-0bedee5fb595.png)

**Comment by macandcheese** [comment]
I agree with this use case - however I don't think it should be "True by default" - disabling the padding here would be a breaking visual change so having it as opt-in behavior works.

cc @asangma - we may also want to look at adding a reciprocal pattern to panel to provide a consistent spacing when needed.

**Comment by TheBlueDog** [comment]
@macandcheese maintaining current behavior as the default is what I meant. The description has been updated to reflect that. Good catch.

**Comment by macandcheese** [comment]
cc @asangma - any issues with this? @benelan - can we re-triage with accurate priority for an upcoming sprint?

**Comment by macandcheese** [comment]
Bump @benelan @jcfranco - seems like a simple one to add - would be great for some Dev site components where I have to set this manually.

**Comment by geospatialem** [comment]
Verified on `next`

### `noPadding` / `no-padding` is `false` (default)
![no padding default verified](https://user-images.githubusercontent.com/5023024/175563668-dda71d3a-bad9-4e02-b0c6-14180abd5e6b.png)

### `noPadding` / `no-padding` is `true`
![no padding set to true verified](https://user-images.githubusercontent.com/5023024/175563610-cfc7717c-0962-4db7-b078-86e28f33b8d4.png)

**Comment by driskull** [comment]
I'm not really sold on the name of the property. 

Would `compact` or `disable-padding` be better?

**Comment by macandcheese** [comment]
`disable-padding` sounds better to me - `no-padding` is currently used on Modal, could be updated there as well.

**Comment by driskull** [comment]
Yeah we should do an audit where we are using `no-`. I think `disable-` sound more professional and aligns with native elements better.

**Comment by anveshmekala** [comment]
I can do the audit for `no-` prefixed props in the consistency wiki. Will use this issue to remove (not deprecate) the `noPadding` attr and add `disable-padding`. This makes it a breaking change but I assume it won't have much impact since it's only been three days.

**Comment by driskull** [comment]
I don't think its a breaking change if we haven't released yet right?

**Comment by anveshmekala** [comment]
> I don't think its a breaking change if we haven't released yet right?

Yupp. My bad.
